rbd-target-api.py

```diff
@@ -2361,44 +2387,18 @@ def get_settings():
     Examples:
     curl --insecure --user admin:admin -X GET https://192.168.122.69:5000/api/settings
     """
-    type_int = "int"
-    type_bool = "bool"
-
```

Comment:
Hey @ricardoasmarques 

I noticed one small possible future issue with the patch that added the types. We seem to have a mix of 2 kinds of bools.

1. LIO based bools. These are iscsi rfc values and use yes/no like defined in that spec and used in the kernel. immediate_data is an example of this type.

2. I guess this would be called ceph-iscsi bools. These are the standard true/false values. api_secure is an example of this kind.

Currently, get_settings only reports about type 1. For this we report type=bool, and then for the values that are reported we use Yes/No. If we ever add support for type 2, would we report type=bool but and for the values would we use True/False?

Note that the PUT api calls can handle true/false, yes/no and 1/0, so we can handle anything. But I was just wondering if for your use case, you want me to separate the 2 types? If so, we could do

1. type=lio_bool (yes/no)
2 type=bool (true/false)

or

bool is always yes/no

or keep what we have:

bool= yes/no or true/false

?

Let me know and I can fix that in this patchset if you want.

Reply:
From the point of view of an external REST API consumer, I think I would prefer to don't have to know about that special type and treat everything as real True/False booleans (and `ceph-iscsi` would do all the conversions internally).

But looking into the output of `/api/settings`:
```
"target_controls_limits": {
        ...
        "immediate_data": {
            "type": "bool"
        },
        ....
    },
    "target_default_controls": {
        ...
        "immediate_data": "Yes",
        ...
    },
```
I see that it would require to return `True` instead of `"Yes"` in the default values.

**Do you think it makes sense to convert the default value to a True/False boolean?** 
If yes, then having a single "bool" type seems good to me.
Otherwise, having a special `lio_bool` seems more correct because it will help the consumer on how to process the default "Yes" value.

Reply:
Ok. I kept bool and made it return true/false, so it is consistent.

yes/no will be used when interacting with lio internally and then we also continue to allow tools/users to pass in yes/no for iscsi RFC values for compatibility.

Let me know if I misunderstood you. My allergies are really bad and so reading is hurting my eyes.